### DISCUSSION TITLE
No 'Access-Control-Allow-Origin' header is present on the requested resource.

### DISCUSSION THREAD

**Discussion by hasnainali9** [vercel/next.js#45964] [Help]
Hey i had a nextjs application which is sending a request to nodejs application. The application also use socket but when i try to send axios request to that nodejs application it gives me CORS error i try to put the header in the request but no luck. Can i know how to fix that.

**Comment by wwwdepot** [comment]
1. If you own "that nodejs application", you can add those missing headers in the response. 
2. Otherwise, use Next.js API routes to "proxy" traffics to "that nodejs application". Server-side fetch does not honor CORS, so it will go through.
`You -> Next.js API Route -> that nodejs application`

**Reply by icyJoseph** [reply]
This is the right answer.

In general CORS errors are because of server configuration, or lack thereof, which is why, owning the "nodejs application", is vital so that you can configure CORS for it. 

Otherwise, you might be able to get around it by adding an extra hop, from Client to Next.js API Routes, and to the "nodejs application".
